lib/theory/src/Theory/Tools/Wellformedness.hs

```diff
@@ -457,6 +476,64 @@ factReports thy = concat
         : (do RuleItem ru <- get thyItems thy; get rActs $ get oprRuleE ru)
           ++ (do RuleItem ru <- get thyItems thy; racs <- get oprRuleAC ru; get rActs racs)
 
+    
+    -- | To bound a fact with his most similar fact
+    -- | The editing distance of two facts must between 1(includ) and 3 (includ)
+    mostSimilarName :: [RuleAndFact]->[RuleAndFact]
+                      ->[(RuleAndFact,RuleAndFact)]
+    mostSimilarName xs xt = 
+        filter (\x-> (notSame (fst x) $ snd x)
+         && (isSimiler (fst x) $ snd x) ) $ 
+        foldr (\x acc-> ((,) x $ minEd (snd x) xt):acc) [] xs
```

Comment:
I guess it would be more efficient to first filter fact that appear on a LHS and not on a RHS, before looking for similar facts?